src/Scenes/GameScene/Renderers/Effects/NextProjectionMoveEffect.ts

```diff
@@ -0,0 +1,83 @@
+import * as PIXI from 'pixi.js';
+import {Effect} from './Effect';
+import {FastPool, Pool} from 'evidently-data-structures';
+import {Protagonist} from '../../../../GameLogic/Entities/Protagonist';
+import Constants from '../../../../Core/Constants';
+import {PlayerActionUtils} from '../../../../GameLogic/Enums';
+import {Direction8Utils} from '../../../../GameLogic/Enums/Direction8';
+import {GfxConstants} from '../../../../Core/Constants/GfxConstants';
+import {TextureStore} from 'evidently-pixi';
+
+const initialAlpha = 0.7;
+const finalAlpha = 0.3;
+const slideLength = 1000;
+const effectBreak = 500;
+const cycleLength = slideLength + effectBreak;
+
+export class NextProjectionMoveEffect extends PIXI.Sprite implements Effect {
+	private static readonly _pool: Pool<NextProjectionMoveEffect>;
+
+	public static getOne(projection: Protagonist, textureStore: TextureStore): NextProjectionMoveEffect {
+		const effect = pool.getOne();
```

Comment:
What is pool here? I only see _pool being defined.

Reply:
Whoops, refactored and didn't check if it still works after the refactor